### ISSUE TITLE
ContrastChecker: audit font size logic 

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#8614]
Follow up to #8059 /Cc @jorgefilipecosta 

I may be wrong but seems to me the font size check is based on pixels and not points. Please notice that the WCAG requirement refers to points, not pixels. Reference:

https://www.w3.org/TR/WCAG21/#contrast-minimum
> Large-scale text and images of large-scale text have a contrast ratio of at least 3:1;

"Large scale text" is defined in points, not pixels:
https://www.w3.org/TR/WCAG21/#dfn-large-scale
> with at least 18 point or 14 point bold 

> The actual size of the character that a user sees is dependent both on the author-defined size and the user's display or user-agent settings. For many mainstream body text fonts, 14 and 18 point is roughly equivalent to 1.2 and 1.5 em or to 120% or 150% of the default size for body text (assuming that the body font is 100%), but authors would need to check this for the particular fonts in use.
etc...

So, calculating the actual _point_ size may depend on various conditions. However, it's an (more or less) accepted practice to use 24 pixels and 18.5 pixels as values. This is explained in the WCAG "Understanding Success Criterion 1.4.3: Contrast (Minimum)":

https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html
> When evaluating this success criterion, the font size in points should be obtained from the user agent or calculated on font metrics in the way that user agents do. Point sizes are based on the CSS pt size as defined in CSS3 Values. The ratio between sizes in points and CSS pixels is 1pt = 1.333px, therefore 14pt and 18pt **are equivalent to approximately 18.5px and 24px**.

Since it would be impossible to use a value in points, the current pixels value of `18` should be changed to `24`.
